support/str/str_test.go

```diff
@@ -4,11 +4,1036 @@ import (
+	s.False(Of("3f2504e0-4f89-41d3-9a0c-0305e82c3301-extra").IsUuid())
+}
+
+func (s *StringTestSuite) TestKebab() {
+	s.Equal("goravel-framework", Of("GoravelFramework").Kebab().String())
+}
+
+func (s *StringTestSuite) TestLcFirst() {
+	s.Equal("framework", Of("Framework").LcFirst().String())
+	s.Equal("framework", Of("framework").LcFirst().String())
+}
+
+func (s *StringTestSuite) TestLength() {
+	s.Equal(11, Of("foo bar baz").Length())
+	s.Equal(0, Of("").Length())
+}
+
+func (s *StringTestSuite) TestLimit() {
+	s.Equal("This is...", Of("This is a beautiful morning").Limit(7).String())
+	s.Equal("This is****", Of("This is a beautiful morning").Limit(7, "****").String())
+	s.Equal("这是一...", Of("这是一段中文").Limit(3).String())
+	s.Equal("这是一段中文", Of("这是一段中文").Limit(9).String())
+}
+
+func (s *StringTestSuite) TestLower() {
+	s.Equal("foo bar baz", Of("FOO BAR BAZ").Lower().String())
+	s.Equal("foo bar baz", Of("fOo Bar bAz").Lower().String())
+}
+
+func (s *StringTestSuite) TestLTrim() {
+	s.Equal("foo ", Of(" foo ").LTrim().String())
+}
+
+func (s *StringTestSuite) TestMask() {
+	s.Equal("kri**************", Of("krishan@email.com").Mask("*", 3).String())
+	s.Equal("*******@email.com", Of("krishan@email.com").Mask("*", 0, 7).String())
+	s.Equal("kris*************", Of("krishan@email.com").Mask("*", -13).String())
+	s.Equal("kris***@email.com", Of("krishan@email.com").Mask("*", -13, 3).String())
+
+	s.Equal("*****************", Of("krishan@email.com").Mask("*", -17).String())
+	s.Equal("*****an@email.com", Of("krishan@email.com").Mask("*", -99, 5).String())
+
+	s.Equal("krishan@email.com", Of("krishan@email.com").Mask("*", 17).String())
+	s.Equal("krishan@email.com", Of("krishan@email.com").Mask("*", 17, 99).String())
+
+	s.Equal("krishan@email.com", Of("krishan@email.com").Mask("", 3).String())
+
+	s.Equal("krissssssssssssss", Of("krishan@email.com").Mask("something", 3).String())
+
+	s.Equal("这是一***", Of("这是一段中文").Mask("*", 3).String())
+	s.Equal("**一段中文", Of("这是一段中文").Mask("*", 0, 2).String())
+}
+
+func (s *StringTestSuite) TestMatch() {
+	s.Equal("World", Of("Hello, World!").Match("World").String())
+	s.Equal("(test)", Of("This is a (test) string").Match(`\([^)]+\)`).String())
+	s.Equal("123", Of("abc123def456def").Match(`\d+`).String())
+	s.Equal("", Of("No match here").Match(`\d+`).String())
+	s.Equal("Hello, World!", Of("Hello, World!").Match("").String())
+	s.Equal("[456]", Of("123 [456]").Match(`\[456\]`).String())
+}
+
+func (s *StringTestSuite) TestMatchAll() {
+	s.Equal([]string{"World"}, Of("Hello, World!").MatchAll("World"))
+	s.Equal([]string{"(test)"}, Of("This is a (test) string").MatchAll(`\([^)]+\)`))
+	s.Equal([]string{"123", "456"}, Of("abc123def456def").MatchAll(`\d+`))
+	s.Equal([]string(nil), Of("No match here").MatchAll(`\d+`))
+	s.Equal([]string{"Hello, World!"}, Of("Hello, World!").MatchAll(""))
+	s.Equal([]string{"[456]"}, Of("123 [456]").MatchAll(`\[456\]`))
+}
+
+func (s *StringTestSuite) TestIsMatch() {
```

Comment:
You said that you use Laravel's test cases, but there are some differences. Is there a specific reason, please?

<img width="834" alt="image" src="https://github.com/goravel/framework/assets/24771476/00fa18f6-90f4-406f-b59c-209f0026cae7">

Reply:
Both are the same. It's just that regex is defined a little differently than PHP.